### ISSUE TITLE
False positive for missing-yield-type-doc in annotated generator

### ISSUE THREAD

**Issue by mwchase** [PyCQA/pylint#3185]
### Steps to reproduce
1. Enable the `pylint.extensions.docparams` plugin in pylintrc
2. Run against the following file:
```
"""Small repro"""

import typing


def generator() -> typing.Iterator[int]:
    """A simple function for reproducing a pylint bug.

    Yields:
        The number 0
    """
    yield 0
```

### Current behavior
```
pylint repro.py 
************* Module repro
repro.py:6:0: W9014: Missing yield type documentation (missing-yield-type-doc)
```

### Expected behavior
No warning.

### pylint --version output
```
pylint 2.4.2
astroid 2.3.1
Python 3.7.4 (default, Aug 27 2019, 22:43:30) 
[Clang 10.0.0 (clang-1000.11.45.5)]
```
